### ISSUE TITLE
Unable to process .mdx files which replaced a previously used .md files in the Gatsby Project

### ISSUE THREAD

**Issue by jnsrikanth** [gatsbyjs/gatsby#21427]
## Description

I had .md (content) files being called from within a .json file in a folder structure under my pages directory in my gatsby project. Basically each page is a pair of the json file combined with a .md file. 

The contents of the JSON file is as follows. 
```
{
  "title": "Technovature Blog",
  "content": "content.md"
}
```
and the contents of the content.md is as below.

```
# Technovature Blog

Amazing Content is getting prepared. Come Back here for some Amazing Content.
```

### Steps to reproduce

Clear steps describing how to reproduce the issue. Please please please link to a demo project if possible, this makes your issue _much_ easier to diagnose (seriously).

How to Make a Minimal Reproduction: https://www.gatsbyjs.org/contributing/how-to-make-a-reproducible-test-case/

Now all I did is renamed the content.md file by simply replacing the .md extension with a .mdx extension and made the below recommended changes to the gatsby-config.js as well as the page queries in the pages folder in order to incorporate the MDX plugins of the GatsbyJS.
I removed all references to the gatsby-plugin-transformer-remark in the gatsby-config.js and replaced with the configuration for the MDX plugin.

The corresponding JS Page that reads the above JSON and processes the .md (or .mdx) file embedded in that folder with the GraphQL query is as follows...

blog.js page contents are below:

```
import React from 'react';
import PropTypes from 'prop-types';
import { graphql } from 'gatsby';
import Layout from 'components/layout';
import Box from 'components/box';
import Head from 'components/head';
import { MDXRenderer } from "gatsby-plugin-mdx";

const Blog = ({ data }) => (
  <Layout>
    <Head pageTitle={data.blogJson.title} />
    <Box>
        <MDXRenderer>{data.blogJson.content.mdx.body}</MDXRenderer>  
    </Box>
  </Layout>
);

Blog.propTypes = {
  data: PropTypes.object.isRequired,
};

export default Blog;

export const query = graphql`
  query BlogQuery {
    blogJson {
      title
      content {
        mdx {
          body
        }
      }
    }
  }
`;
```

gatsby-config.js file contents are as below:

```
const siteConfig = require('./site-config');

module.exports = {
  siteMetadata: {
    ...siteConfig,
  },
  plugins: [
    `gatsby-plugin-react-helmet`,
    `gatsby-plugin-sitemap`,
    `gatsby-plugin-offline`,
    `gatsby-transformer-json`,
    `gatsby-plugin-eslint`,
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `content`,
        path: `${__dirname}/content`,
      },
    },
    {
      resolve: `gatsby-plugin-mdx`,
      options: {
        extensions: [`.mdx`, `.md`],
        gatsbyRemarkPlugins: [
      {
        resolve: `gatsby-remark-images`,
        options: {
          maxWidth: 590,
        },
      },
      {
        resolve: `gatsby-remark-responsive-iframe`,
        options: {
          wrapperStyle: `margin-bottom: 1.0725rem`,
        },
      },
      `gatsby-remark-prismjs`,
      `gatsby-remark-copy-linked-files`,
      `gatsby-remark-smartypants`,
      ],
    },
    },
    `gatsby-plugin-sharp`,
    `gatsby-transformer-sharp`,
    `gatsby-plugin-webpack-size`,
    {
      resolve: `gatsby-plugin-react-svg`,
      options: {
        rule: {
          include: /images\/.*\.svg$/,
        },
      },
    },
    {
      resolve: `gatsby-plugin-typography`,
      options: {
        pathToConfigModule: `src/utils/typography`,
      },
    },
    
  ],
};
```

gatsby-node.js content is as follows:

```
const path = require('path');
const DirectoryNamedWebpackPlugin = require('directory-named-webpack-plugin');

exports.onCreateWebpackConfig = ({
  stage,
  getConfig,
  rules,
  loaders,
  actions,
}) => {
  actions.setWebpackConfig({
    resolve: {
      modules: [path.resolve(__dirname, 'src'), 'node_modules'],
      plugins: [new DirectoryNamedWebpackPlugin({
        exclude: /node_modules/
      })],
    },
  });
};
```

### Expected result

Expected a proper processing of MDX files. 

### Actual result

Got the below error.

8:55:17 PM: export GIMME_ENV='/opt/buildhome/.gimme/env/go1.12.linux.amd64.env';
8:55:17 PM: go version go1.12 linux/amd64
8:55:17 PM: Installing missing commands
8:55:17 PM: Verify run directory
8:55:17 PM: Executing user command: gatsby build
8:55:20 PM: success open and validate gatsby-configs — 0.013 s
8:55:22 PM: success load plugins — 1.700 s
8:55:22 PM: success onPreInit — 0.007 s
8:55:22 PM: success delete html and css files from previous builds — 0.006 s
8:55:22 PM: success initialize cache — 0.017 s
8:55:22 PM: success copy gatsby files — 0.023 s
8:55:22 PM: success onPreBootstrap — 0.011 s
8:55:24 PM: success source and transform nodes — 2.417 s
8:55:24 PM: success building schema — 0.267 s
8:55:24 PM: success createPages — 0.025 s
8:55:25 PM: success createPagesStatefully — 0.039 s
8:55:25 PM: success onPreExtractQueries — 0.003 s
8:55:25 PM: success update schema — 0.025 s
8:55:25 PM: error GraphQL Error Field "content" must not have a selection since type "String" has no subfields.
8:55:25 PM:   file: /opt/build/repo/src/pages/about.js
8:55:25 PM:    1 |
8:55:25 PM:    2 |   query AboutQuery {
8:55:25 PM:    3 |     aboutJson {
8:55:25 PM:    4 |       title
8:55:25 PM: >  5 |       content {
8:55:25 PM:      |               ^
8:55:25 PM:    6 |         mdx {
8:55:25 PM:    7 |           body
8:55:25 PM:    8 |         }
8:55:25 PM:    9 |       }
8:55:25 PM:   10 |     }
8:55:25 PM:   11 |   }
8:55:25 PM:   12 |
8:55:25 PM: Skipping functions preparation step: no functions directory set

So what am I missing here? Please help.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.
System:
    OS: Windows 7 6.1.7601
    CPU: (4) x64 Intel(R) Core(TM) i5 CPU       M 560  @ 2.67GHz
  Binaries:
    Node: 12.14.1 - D:\Program Files\nodejs\node.EXE
    npm: 6.13.4 - D:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 2.7.17 - /c/Python27/python
  npmPackages:
    gatsby: ^2.3.26 => 2.19.16
    gatsby-image: ^2.0.39 => 2.2.40
    gatsby-plugin-eslint: ^2.0.3 => 2.0.8
    gatsby-plugin-mdx: ^1.0.71 => 1.0.71
    gatsby-plugin-offline: ^2.0.25 => 2.2.10
    gatsby-plugin-react-helmet: ^3.0.12 => 3.1.22
    gatsby-plugin-react-svg: ^2.0.0 => 2.1.2
    gatsby-plugin-sharp: ^2.0.35 => 2.4.5
    gatsby-plugin-sitemap: ^2.0.12 => 2.2.27
    gatsby-plugin-styled-components: ^3.0.7 => 3.1.19
    gatsby-plugin-typography: ^2.3.22 => 2.3.22
    gatsby-plugin-webpack-size: ^0.0.3 => 0.0.3
    gatsby-remark-copy-linked-files: ^2.1.37 => 2.1.37
    gatsby-remark-images: ^3.1.44 => 3.1.44
    gatsby-remark-prismjs: ^3.3.31 => 3.3.31
    gatsby-remark-responsive-iframe: ^2.2.32 => 2.2.32
    gatsby-remark-smartypants: ^2.1.21 => 2.1.21
    gatsby-source-filesystem: ^2.0.30 => 2.1.48
    gatsby-transformer-json: ^2.1.11 => 2.2.26
    gatsby-transformer-remark: ^2.3.9 => 2.6.50
    gatsby-transformer-sharp: ^2.1.18 => 2.3.14

**Comment by jonniebigodes** [comment]
@jnsrikanth i'll take a look at this starting tomorrow and during the weekend and come back possibly with a response and solution to your issue. Do you mind waiting a bit?

**Comment by jnsrikanth** [comment]
@jonniebigodes - I will be vey happy to see a solution from you. I will surely wait for the same. Thanks.

**Comment by LekoArts** [comment]
Hi!

Sorry to hear you're running into an issue. To help us best begin debugging the underlying cause, it is incredibly helpful if you're able to create a [minimal reproduction][reproduction]. This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

If you're up for it, we'd very much appreciate if you could [provide a minimal reproduction][reproduction] and we'll be able to take another look.

Thanks for using Gatsby! 💜

[reproduction]: https://gatsby.dev/reproduction

**Comment by jnsrikanth** [comment]
Hello @LekoArts I am providing you with a public copy of my minimal repository here. 

https://github.com/jnsrikanth/gatsby-universal

Let me know, if you need any additional info.

Thanks,
Srikanth

**Comment by jnsrikanth** [comment]
@LekoArts made the repository pivate and added you for repository access. Please check.

**Comment by jnsrikanth** [comment]
@jonniebigodes and @LekoArts -  any luck with my issue getting some headway? 

Thanks,
jnsrikanth

**Comment by jonniebigodes** [comment]
@jnsrikanth i think i'm on the right track. I'm going to run some more tests and come back with a response. Sounds good?

**Comment by jnsrikanth** [comment]
@jonniebigodes yes json gives me a big advantage to source catalog or content externally whereas mdx provides me an ability to display rich content. Also I want to be able to use JSON to create a structure and relationships within my site while MDX is what becomes visible to the outside world in the form of rich content. Hope I have answered it enough to make sense?

**Comment by jnsrikanth** [comment]
@jonniebigodes I hope I was able to set you up in looking into my issue with the details I provided?
Thanks.

**Comment by jnsrikanth** [comment]
@jonniebigodes , @LekoArts  - I am pinging you both to see if there was any breakthrough made in the issue I reported. Any pointers will be highly appreciated. Thanks.

**Comment by jonniebigodes** [comment]
@jnsrikanth sorry for the radio silence. It's not that i've forgotten your issue. But i've been trying to tackle this from every angle i can think of and i cannot seem to make it work. I've got to the point of "throwing the kitchen sink"(pardon the bad pun) to see if both plugins could work together and to no avail. What gets me baffled everytime is that if i throw a image, a csv file or or any other file that has a plugin i can make  it work. I get what i need. I went to the extreme of manually forcing the schema to see if i could overcome this and i wasn't able to. I'm trully sorry that i couldn't help you further, probably @LekoArts has more insights on this and could help you further with your issue

**Comment by jonniebigodes** [comment]
@jnsrikanth sorry for the radio silence. It's not that i've forgotten your issue. But i've been trying to tackle this from every angle i can think of and i cannot seem to make it work. I've got to the point of "throwing the kitchen sink"(pardon the bad pun) to see if both plugins could work together and to no avail. What gets me baffled everytime is that if i throw a image, a csv file or or any other file that has a plugin i can make  it work. I get what i need. I went to the extreme of manually forcing the schema to see if i could overcome this and i wasn't able to. I'm trully sorry that i couldn't help you further, probably @LekoArts has more insights on this and could help you further with your issue
